src/averageAges.js

```diff
@@ -55,7 +77,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childMotherPairs = people.map((child) => {
+    const mother = (!onlyWithSon || child.sex === 'm')
+      && child.mother
+      && people.find((person) => (
+        person.name === child.mother)
+      );
+
+    return mother && [child.born, mother.born];
+  });
+
+  const ageDiff = childMotherPairs
+    .filter(pair => pair)
```

Comment:
Instead of this, better filter only mothers/children, then map the filtered array and add property f.e. `motherBorn`/`childBorn`.

To make your `getAverage` helper more universal pass two additional args: `keyA`, `keyB`. So, inside reducer, you'll be able to use them: `sum + person[keyA] - person[keyB]`.